### PR TITLE
Implemented configurable minimum signup age

### PR DESCRIPTION
**Pull request by Poslovitch**
## Description

I added a config option to set the minimum age required in order to signup to the instance. It defaults to 16 (which has been the "default" age on PeerTube up to now).
I therefore modified the RegisterComponent so that it can display this age.

## Related issues

Resolves https://github.com/Chocobozzz/PeerTube/issues/3612.

## Has this been tested?

- [ ] 👍 yes, I added tests to the test suite
- [x] 👍 yes, light tests as follows are enough
- [ ] 💭 no, because this PR is a draft and still needs work
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

I believe light tests were enough, but should tests be added to the test suite?
If so:
* On which test suit?
* Which tests scenarios?

## Screenshots

In the sign up page:
image

In the admin "Basic configuration" editor :
image

### CODE REVIEW

File: client/src/app/+admin/config/edit-custom-config/edit-basic-configuration.component.html [hunk 1]
```diff
@@ -478,7 +492,7 @@
         <ng-container formGroupName="twitter">
 
           <div class="form-group">
-            <label i18n for="signupLimit">Your Twitter username</label>
```
Comment (Chocobozzz) [inline client/src/app/+admin/config/edit-custom-config/edit-basic-configuration.component.html:481]: Oup's, thanks. It should be `for="servicesTwitterUsername"` instead

File: client/src/app/+signup/+register/register-step-terms.component.html [hunk 2]
```diff
@@ -2,8 +2,8 @@
   <div class="form-group form-group-terms">
     <my-peertube-checkbox inputName="terms" formControlName="terms">
       <ng-template ptTemplate="label">
-        <ng-container i18n>
-          I am at least 16 years old and agree
+        <ng-container i18n> 
+          I am at least {{ minimumAge }} years old and agree
```
Comment (Chocobozzz) [inline client/src/app/+signup/+register/register-step-terms.component.html:6]: Please remove added spaces at the end of these lines

File: client/src/app/+signup/+register/register.component.html [hunk 3]
```diff
@@ -17,6 +17,7 @@
 
           <my-register-step-terms
             [hasCodeOfConduct]="!!aboutHtml.codeOfConduct"
+            [minimumAge]="this.minimumAge"
```
Comment (Chocobozzz) [inline client/src/app/+signup/+register/register.component.html:20]:
```suggestion
            [minimumAge]="minimumAge"
```

File: config/default.yaml [hunk 4]
```diff
@@ -229,6 +229,7 @@ contact_form:
 signup:
   enabled: false
   limit: 10 # When the limit is reached, registrations are disabled. -1 == unlimited
+  minimum_age: 16
```
Comment (Chocobozzz) [inline config/default.yaml:232]: Please add a comment to explain this is used in the signup form

File: shared/extra-utils/server/config.ts [hunk 5]
```diff
@@ -98,7 +98,8 @@ function updateCustomSubConfig (url: string, token: string, newConfig: DeepParti
     signup: {
       enabled: false,
       limit: 5,
-      requiresEmailVerification: false
+      requiresEmailVerification: false,
+      minimumAge: 16
```
Comment (Chocobozzz) [inline shared/extra-utils/server/config.ts:102]: Please use another value and check it is correctly set with the default in https://github.com/Chocobozzz/PeerTube/blob/develop/server/tests/api/server/config.ts#L63 and correctly updated in https://github.com/Chocobozzz/PeerTube/blob/develop/server/tests/api/server/config.ts#L153

**Review by Chocobozzz — CHANGES_REQUESTED** [review]
Thanks!

File: client/src/app/+signup/+register/register-step-terms.component.html (hunk 2 above)
Comment (rigelk) [reply client/src/app/+signup/+register/register-step-terms.component.html:6]:
```suggestion
          I am at least {{ minimumAge }} years old and agree
```

File: config/default.yaml (hunk 4 above)
Comment (rigelk) [reply config/default.yaml:232]:
```suggestion
  minimum_age: 16 # Used to configure the signup form
```

**Comment by Poslovitch** [comment]
Quick heads-up : I'll provide the requested changes this week.

**Comment by Chocobozzz** [comment]
Merged manually in https://github.com/Chocobozzz/PeerTube/commit/1f256e7d3cf056c2d999260155cdba58ae1b878b

**Pull request closed without merging**